// postcss-selector-parser/dist/selectors/selectors.js
'use strict';

Object.defineProperty(exports, '__esModule', {
    value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

var Selectors = (function () {
    function Selectors() {
        _classCallCheck(this, Selectors);

        this.type = 'selectors';
        this.rules = [];
        this.spaces = { before: '', after: '' };
    }

    _createClass(Selectors, [{
        key: 'append',
        value: function append(selector) {
            selector.parent = this;
            this.rules.push(selector);
            return this;
        }
    }, {
        key: 'each',
        value: function each(callback) {
            this.rules.forEach(callback);
            return this;
        }
    }, {
        key: 'toString',
        value: function toString() {
            return [this.spaces.before, this.rules.map(String).join(''), this.spaces.after].join('');
        }
    }]);

    return Selectors;
})();

exports['default'] = Selectors;
module.exports = exports['default'];
